lib/SILOptimizer/IPO/GlobalOpt.cpp

```diff
@@ -637,15 +641,26 @@ static SILFunction *genGetterFromInit(SILOptFunctionBuilder &FunctionBuilder,
   return GetterF;
 }
 
-/// Checks if a given global variable is assigned only once.
-static bool isAssignedOnlyOnceInInitializer(SILGlobalVariable *SILG) {
+bool SILGlobalOpt::isAssignedOnlyOnceInInitializer(SILGlobalVariable *SILG,
+                                                   SILFunction *globalAddrF) {
   if (SILG->isLet())
     return true;
-  // TODO: If we can prove that a given global variable
-  // is assigned only once, during initialization, then
-  // we can treat it as if it is a let.
-  // If this global is internal or private, it should be
-  return false;
+
+  // If we should skip this, it is probably because there are multiple stores.
+  // Return false if there are multiple stores or no stores.
+  if (GlobalVarSkipProcessing.count(SILG) || !GlobalVarStore.count(SILG) ||
+      // Check if there is more than one use the global addr function. If there
+      // is only one use, it must be the use that we are trying to optimize, so
+      // that is OK. If there is more than one use, one of the other uses may
+      // have a store attached to it which means there may be more than one
+      // assignment, so return false.
```

Comment:
I added this additional check to prevent an edge case where global uses were improperly optimized away. In the future, `GlobalInitCallMap` applies could theoretically be traced to a store instruction that could be added to `GlobalVarStore` which would be beneficial for a few reasons (including that we could get rid of this extra check and optimize more code correctly).